trieste/models/keras/models.py

```diff
@@ -359,15 +367,82 @@ def optimize(self, dataset: Dataset) -> None:
+
+        # Tell optimizer how many epochs have been used before: the optimizer will "continue"
+        # optimization across multiple BO iterations rather than start fresh at each iteration.
+        # This allows us to monitor training across iterations.
+
+        if "epochs" in fit_args:
+            fit_args["epochs"] = fit_args["epochs"] + self._absolute_epochs
 
         x, y = self.prepare_dataset(dataset)
-        self.model.fit(x=x, y=y, **self.optimizer.fit_args)
+        history = self.model.fit(
+            x=x,
+            y=y,
+            **fit_args,
+            initial_epoch=self._absolute_epochs,
+        )
+        if self._continuous_optimisation:
+            self._absolute_epochs = self._absolute_epochs + len(history.history["loss"])
 
         # Reset lr in case there was an lr schedule: a schedule will have change the learning rate,
         # so that the next time we call `optimize` the starting learning rate would be different.
         # Therefore, we make sure the learning rate is set back to its initial value.
         self.optimizer.optimizer.lr.assign(self.original_lr)
 
+    def log(self, dataset: Optional[Dataset] = None) -> None:
+        """
+        Log basic model training information at a given optimization step. We log several summary
+        statistics of losses and metrics given in ``fit_args`` to ``optimizer``
+        (final, difference between inital and final loss, min and max), for the whole ensemble and
+        individual models. We also log epoch statistics, but as histograms, rather than time series.
+
+        We also log several training data based metrics, such as root mean square error and mean
+        aboslute error between predictions and observations, as well as variances for predictions
+        (means and histograms of individual values).
+
+        For custom logs user will need to subclass the model and overwrite this method.
+
+        :param dataset: Optional data that can be used to log additional data-based model summaries.
+        """
+        summary_writer = logging.get_tensorboard_writer()
+        if summary_writer:
+            with summary_writer.as_default(step=logging.get_step_number()):
+                logging.scalar("num_epochs", len(self.model.history.epoch))
+                for k, v in self.model.history.history.items():
+                    logging.histogram(f"{k}_epoch", v)
+                    logging.scalar(f"{k}_final", v[-1])
+                    logging.scalar(f"{k}_diff", v[0] - v[-1])
+                    logging.scalar(f"{k}_min", tf.reduce_min(v))
+                    logging.scalar(f"{k}_max", tf.reduce_max(v))
+                if dataset:
+                    predict = self.predict(dataset.query_points)
+                    # training accuracy
+                    diffs = predict[0] - tf.cast(dataset.observations, predict[0].dtype)
+                    logging.histogram("accuracy_absolute_errors", tf.math.abs(diffs))
```

Comment:
Again, I'd collate all `accuracy` logs together, and similarly for `variance` logs.

```suggestion
                    logging.histogram("accuracy/absolute_errors", tf.math.abs(diffs))
```

Reply:
How about logging the histogram of the normalised residuals? You'd get a better indication of the behaviour of the model independent on the scale of the differences.
On a similar note, you could log the Z-score of the normalised residuals (should be close to 1).

Reply:
good idea, I have added 
- z residuals histogram
- variance error histogram
- z residuals std scalar
- root mean variance error scalar